src/mysqlsh_helpers.py

```diff
@@ -340,6 +356,129 @@ def remove_instances_not_online(self) -> None:
+            username: The username of the user to create
+            password: THe user's password
+            label: The label to tag the user with (to be able to delete it later)
+            hostname: (Optional) The hostname of the new user to create (% by default)
+        """
+        try:
+            primary_address = self.get_cluster_primary_address()
+
+            escaped_user_attributes = json.dumps({"label": label}).replace('"', r"\"")
+            create_user_commands = (
+                f"shell.connect('{self.server_config_user}:{self.server_config_password}@{primary_address}')",
+                f"session.run_sql(\"CREATE USER '{username}'@'{hostname}' IDENTIFIED BY '{password}' ATTRIBUTE '{escaped_user_attributes}';\")",
+            )
+
+            self._run_mysqlsh_script("\n".join(create_user_commands))
+        except MySQLClientError as e:
+            logger.exception(f"Failed to create user {username}@{hostname}", exc_info=e)
+            raise MySQLCreateUserError(e.message)
+
+    def escalate_user_privileges(self, username: str, hostname: str = "%") -> None:
+        """Escalates the provided user's privileges.
+
+        Args:
+            username: The username of the user to escalate privileges for
+            hostname: The hostname of the user to escalate privileges for
+        """
+        try:
+            super_privileges_to_revoke = (
+                "SYSTEM_USER",
+                "SYSTEM_VARIABLES_ADMIN",
+                "SUPER",
+                "REPLICATION_SLAVE_ADMIN",
+                "GROUP_REPLICATION_ADMIN",
+                "BINLOG_ADMIN",
+                "SET_USER_ID",
+                "ENCRYPTION_KEY_ADMIN",
+                "VERSION_TOKEN_ADMIN",
+                "CONNECTION_ADMIN",
+            )
+
+            primary_address = self.get_cluster_primary_address()
+
+            escalate_user_privileges_commands = (
+                f"shell.connect('{self.server_config_user}:{self.server_config_password}@{primary_address}')",
+                f"session.run_sql(\"GRANT ALL ON *.* TO '{username}'@'{hostname}' WITH GRANT OPTION;\")",
+                f"session.run_sql(\"REVOKE {', '.join(super_privileges_to_revoke)} ON *.* FROM '{username}'@'{hostname}';\")",
+                'session.run_sql("FLUSH PRIVILEGES;")',
+            )
+
+            self._run_mysqlsh_script("\n".join(escalate_user_privileges_commands))
+        except MySQLClientError as e:
+            logger.exception(
+                f"Failed to escalate user privileges for {username}@{hostname}", exc_info=e
+            )
+            raise MySQLEscalateUserPrivilegesError(e.message)
+
+    def delete_users_with_label(self, label_name: str, label_value: str) -> None:
+        """Delete users for a unit.
+
+        Args:
+            unit_name: The name of the unit for which to delete mysql users for
+
+        Raises:
+            MySQLDeleteUsersForUnitError if there is an error deleting users for the unit
+        """
```

Comment:
The docstring needs to be updated.

Reply:
addressed in 2f6e6e434f2bab7f2a508153eb234fb73edf2bed

Reply:
It seems that it was not updated in that commit.

Reply:
i believe i updated it [here](https://github.com/canonical/mysql-k8s-operator/pull/88/files#diff-6e566a7b93efc3f3e7ac0e6d8989256725f876f06af78c45d0ae3815d4bcf82eR454) -> the name of the function is slightly different than the copy pasted one

Reply:
Ah, sorry Shayan! I haven't made it clear that more parts of the docstring need changes (the args section and maybe the initial description can be updated; the [args section](https://github.com/canonical/mysql-k8s-operator/commit/2f6e6e434f2bab7f2a508153eb234fb73edf2bed#diff-6e566a7b93efc3f3e7ac0e6d8989256725f876f06af78c45d0ae3815d4bcf82eR451) is not matching the list of arguments of this method)

Reply:
ah sorry, i completely missed that. corrected in f375b063af5362d0b622dce487e68f2bfd54a630